x/mongo/driver/auth/scram_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package auth
+
+import (
+	"context"
+	"testing"
+
+	"go.mongodb.org/mongo-driver/internal/testutil/assert"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/description"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/drivertest"
+)
+
+const (
+	scramSha1Nonce   = "fyko+d2lbbFgONRv9qkxdawL"
+	scramSha256Nonce = "rOprNGfwEbeRWgbNEkqO"
+)
+
+var (
+	scramSha1ShortPayloads = [][]byte{
+		[]byte("r=fyko+d2lbbFgONRv9qkxdawLHo+Vgk7qvUOKUwuWLIWg4l/9SraGMHEE,s=rQ9ZY3MntBeuP3E1TDVC4w==,i=10000"),
+		[]byte("v=UMWeI25JD1yNYZRMpZ4VHvhZ9e0="),
+	}
+	scramSha256ShortPayloads = [][]byte{
+		[]byte("r=rOprNGfwEbeRWgbNEkqO%hvYDpWUa2RaTCAfuxFIlj)hNlF$k0,s=W22ZaJ0SNY7soEsUEjb6gQ==,i=4096"),
+		[]byte("v=6rriTRBi23WpRR/wtup+mMhUZUn/dB5nLTJRsjl95G4="),
+	}
+	scramSha1LongPayloads   = append(scramSha1ShortPayloads, []byte{})
+	scramSha256LongPayloads = append(scramSha256ShortPayloads, []byte{})
+)
+
+func TestSCRAM(t *testing.T) {
```

Comment:
Can this also assert that the driver always appends the skipEmptyExchange option to saslStart?

Does this test assert anything about the client commands being sent?

Reply:
Added the functionality to parse the command that was sent and an assertion that it has the options we expect.